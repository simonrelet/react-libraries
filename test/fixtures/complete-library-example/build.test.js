'use strict'

const path = require('path')
const { cleanFixture, listFolderContent } = require('../../fixtureUtils')
const { callScriptInPackage } = require('../../scriptExecution')
const packageJSON = require('./package.json')

describe(`${packageJSON.name} build`, () => {
  beforeAll(async () => {
    await cleanFixture(__dirname)
  })

  it('should not fail', async () => {
    const result = await callScriptInPackage(__dirname, 'build')
    expect(result).toMatchSnapshot()
  })

  it('should generate all files', async () => {
    const files = await listFolderContent(path.join(__dirname, 'build'))
    expect(files).toMatchSnapshot()
  })
})
