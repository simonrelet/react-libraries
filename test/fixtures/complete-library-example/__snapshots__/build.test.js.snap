// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`complete-library-example build should generate all files 1`] = `
Array [
  "cjs/Button/Button.js",
  "cjs/Button/index.js",
  "cjs/index.js",
  "complete-library-example.css",
  "es/Button/Button.js",
  "es/Button/index.js",
  "es/index.js",
]
`;

exports[`complete-library-example build should not fail 1`] = `
Exit code: 0

--- Output ---
CommonJS build successful.
ES modules build successful.
SASS build successful.
--------------
`;
